File: dynamic_listings/python/default/design_patterns/flyweight.py
class Common:
    # Contains the common data for a 3D Object to be replicated
    def __init__(self, mesh: Mesh, texture: Texture):
        self._mesh: Mesh = mesh
        self._texture: Texture = texture


class FlyWeight:
    # Contains only the necessary data to create an instance of the item
    def __init__(self, common: Common, position: Vector2, scale_factor: float):
        self._common_pointer: Common = common
        self._position: Vector2 = position
        self._scale_factor: float = scale_factor

File: dynamic_listings/rust/default/design_patterns/flyweight.rs
# #![allow(dead_code)]
# struct Mesh;
# struct Texture;
# struct Vector2D;
struct Common {
    mesh: Mesh,
    texture: Texture,
}

struct FlyWeight {
    common_pointer: Common,
    position: Vector2D,
    scale_factor: f32,
}
